scripts/convert_blipdiffusion_to_diffusers.py

```diff
@@ -0,0 +1,169 @@
+    rename_keys.append((f"visual_encoder.encoder.layers.{i}.layer_norm2.weight", f"blip.visual_encoder.transformer.resblocks.{i}.ln_2.weight"))
+    rename_keys.append((f"visual_encoder.encoder.layers.{i}.layer_norm2.bias", f"blip.visual_encoder.transformer.resblocks.{i}.ln_2.bias"))
+    rename_keys.append((f"visual_encoder.encoder.layers.{i}.self_attn.qkv.weight", f"blip.visual_encoder.transformer.resblocks.{i}.attn.in_proj_weight"))
+    rename_keys.append((f"visual_encoder.encoder.layers.{i}.self_attn.qkv.bias", f"blip.visual_encoder.transformer.resblocks.{i}.attn.in_proj_bias"))
+    rename_keys.append((f"visual_encoder.encoder.layers.{i}.self_attn.projection.weight", f"blip.visual_encoder.transformer.resblocks.{i}.attn.out_proj.weight",))
+    rename_keys.append((f"visual_encoder.encoder.layers.{i}.self_attn.projection.bias", f"blip.visual_encoder.transformer.resblocks.{i}.attn.out_proj.bias"))
+    rename_keys.append((f"visual_encoder.encoder.layers.{i}.mlp.fc1.weight", f"blip.visual_encoder.transformer.resblocks.{i}.mlp.c_fc.weight"))
+    rename_keys.append((f"visual_encoder.encoder.layers.{i}.mlp.fc1.bias", f"blip.visual_encoder.transformer.resblocks.{i}.mlp.c_fc.bias"))
+    rename_keys.append((f"visual_encoder.encoder.layers.{i}.mlp.fc2.weight", f"blip.visual_encoder.transformer.resblocks.{i}.mlp.c_proj.weight"))
+    rename_keys.append((f"visual_encoder.encoder.layers.{i}.mlp.fc2.bias", f"blip.visual_encoder.transformer.resblocks.{i}.mlp.c_proj.bias"))
+
+rename_keys.append(("visual_encoder.post_layernorm.weight", "blip.ln_vision.weight"))
+rename_keys.append(("visual_encoder.post_layernorm.bias", "blip.ln_vision.bias"))
+
+
+renaming_dict = {}
+
+for x in rename_keys:
+    renaming_dict[x[0]] = x[1]
+
+for name, param in qformer.named_parameters():
+    if name in renaming_dict:
+        try:
+            qformer.state_dict()[name].copy_(model.state_dict()[renaming_dict[name]])
+        except Exception as e:
+            print(f"Error copying {name} ")
+            print(e)
+
+    else:
+        pass
+        print(f"{name} not found in qformer")
+
+qformer.eval()
+text_encoder = CtxCLIPTextModel.from_pretrained(
+    "runwayml/stable-diffusion-v1-5", subfolder="text_encoder"
+)
+vae = AutoencoderKL.from_pretrained(
+    "runwayml/stable-diffusion-v1-5", subfolder="vae"
+)
+
+unet = UNet2DConditionModel.from_pretrained(
+    "runwayml/stable-diffusion-v1-5", subfolder="unet"
+
+)
+
+vae.eval()
+text_encoder.eval()
+
+
+scheduler = PNDMScheduler(
+    beta_start=0.00085,
+    beta_end=0.012,
+    beta_schedule="scaled_linear",
+    set_alpha_to_one=False,
+    skip_prk_steps=True,
+)
+
+#TODO: Test this once
+tokenizer = CLIPTokenizer.from_pretrained(
+    "./runwayml/stable-diffusion-v1-5/", subfolder="tokenizer", cache_dir='./cache'
+)
+blipDiffusion = BlipDiffusionPipeline(tokenizer=tokenizer, text_encoder=text_encoder,  vae=vae, unet=unet, scheduler=scheduler, qformer=qformer)
+blipDiffusion.save_pretrained("blip_diffusion")
```

Comment:
Could we make it a nice function following the structure of the rest of the scripts present inside `scripts/`?

Reply:
Done, some funny stuff is happening an there seems to be a difference in outputs now